subsys/bluetooth/mesh/settings.c

```diff
@@ -684,6 +698,58 @@ static int vnd_mod_set(const char *name, size_t len_rd,
 	return mod_set(true, name, len_rd, read_cb, cb_arg);
 }
 
+#if defined(CONFIG_BT_MESH_PROVISIONER)
+static int node_set(const char *name, size_t len_rd,
+		    settings_read_cb read_cb, void *cb_arg)
+{
+	struct bt_mesh_node *node;
+	struct node_val val;
+	u16_t addr;
+	int err;
+
+	if (!name) {
+		BT_ERR("Insufficient number of arguments");
+		return -ENOENT;
+	}
+
+	addr = strtol(name, NULL, 16);
+
+	if (len_rd == 0) {
+		BT_DBG("val (null)");
+		BT_DBG("Deleting node 0x%04x", addr);
+
+		node = bt_mesh_node_find(addr);
+		if (node) {
+			bt_mesh_node_del(node, false);
```

Comment:
Is there a scenario where there's a need to do cleanup on load?

Reply:
Please elaborate :)

Reply:
Isn't that what's happening here? I'm asking whether it's actually necessary.

Reply:
Oh I see :p

Well I guess there could be a need if a node has been removed. Currently there is no way to remove a node but I think it makes sense to add that possibility. I'm just talking about removing the node from the list, not doing the whole network key changing thingy.

If I understand the settings correctly if a node has been added and then removed, `node_set()` will be called once for the first entry, which will allocate a new node. Then `node_set()` will be called again but with len 0 meaning it has been removed.

Reply:
Now that #19541 has been merged, I don't think you will get history replay anymore. So in new settings handlers I don't think we need to handle that case (we should also gradually start removing unnecessary code from the existing handlers)